src/utils/daos_hdlr.c

```diff
@@ -2657,159 +2656,103 @@ static int fs_copy_parse_uuids(char *str,
 }
 
 static int
-fs_copy_parse(char *src,
-	      char *dst,
-	      struct fs_copy_args *fa,
-	      struct file_dfs *src_file_dfs,
-	      struct file_dfs *dst_file_dfs,
-	      struct duns_attr_t *src_dattr,
-	      struct duns_attr_t *dst_dattr,
-	      char **src_str,
-	      char **src_dfs_path,
-	      int *src_str_len,
-	      char **dst_str,
-	      char **dst_dfs_path,
-	      int *dst_str_len)
-{
-	/* parse src and dst strings
+fs_copy_parse(struct file_dfs *file_dfs,
+	      struct duns_attr_t *dattr,
+	      char **dfs_path,
+	      char *arg,
+	      char **arg_str,
+	      int *str_len,
+	      uuid_t *p_uuid,
+	      uuid_t *c_uuid)
+{
+	/* parse argument string
 	 * for DAOS, daos: specified in string
 	 * for POSIX, just use path
 	 * example: --src={daos://pool/cont}
 	 * example: --dst={/a/path}
 	 */
 	int	rc = 0;
-	char	*src_saveptr = NULL;
-	char	*dst_saveptr = NULL;
-	char	*src_type = NULL;
-	char	*dst_type = NULL;
-	char	*src_pool = NULL;
-	char	*src_cont = NULL;
-	char	*dst_pool = NULL;
-	char	*dst_cont = NULL;
+	char	*arg_saveptr = NULL;
+	char	*arg_type = NULL;
+	char	*pool = NULL;
+	char	*cont = NULL;
+	int	len = 0;
+	int	pool_cont_len = 0;
 
 	/* validate and enforce "daos://" prefix is used, if not,
 	 * set to "path" type
 	 */
-	if (strncmp(src, "daos://", 7) == 0) {
-		src_type = "daos";
-		/* start of src pool/cont str */
-		src += 7;
+	if (strncmp(arg, "daos://", 7) == 0) {
```

Comment:
I'm not familiar with what this is doing, but I do wonder if duns_resolve_path() would help here, it already parses daos://<uuid>/<uuid> which from the comments is what whis is doing.

Reply:
@ashleypittman I don't think that was available when I started writing this, but I moved to using the duns_resolve_path call to get the rel_path, so that should resolve most of your comments on the parsing below.